### DISCUSSION TITLE
Missing PRs for multiple packages with same or similar name

### DISCUSSION THREAD

**Discussion by lauraseidler** [renovatebot/renovate#10034] [Request Help] [answered]
I have a repository with the (somewhat simplified) following folder and file structure:
```
backend
- Dockerfile
- package.json
frontend
- Dockerfile
- package.json
docker-compose.yaml
```

I noticed now that we are missing some PRs for some packages, and it looks to be somehow related to the package name. This is the files reduced to the problematic dependencies:

`backend/Dockerfile`:
```dockerfile
FROM node:16.1.0-alpine
```

`backend/package.json`:
```json
{
    "devDependencies": {
        "@types/node": "14.14.45"
    }
}
```

`frontend/Dockerfile`:
```dockerfile
FROM node:16.1.0-alpine
```
`frontend/package.json`:
```json
{
    "devDependencies": {
        "@types/node": "12.20.13"
    }
}
```

`docker-compose.yaml`:
```yaml
---
version: "3.7"

services:
  frontend:
    image: node:16.1.0-alpine

  backend:
    image: node:16.1.0-alpine
```

`renovate.json` (shared config, removing some settings that don't apply to this repository):
```json
{
    "extends": ["config:base", "docker:enableMajor"],
    "labels": ["renovate"],
    "separateMinorPatch": true,
    "timezone": "Europe/Berlin"
}
```

**What works fine:**
- the node docker image gets updated, with the update for both Dockerfiles and the docker-compose in the same PR
- `@types/node` in `backend/package.json` gets updated within the `14.x` range
- `@types/node` in `frontend/package.json` gets updated within the `12.x` range
- major version updates for other npm packages

**What doesn't work:**
- `@types/node` in `backend/package.json` is not updated to the `15.x` range
- `@types/node` in `frontend/package.json` is not updated to the `15.x` range, and also was not updated to the `14.x` range even before `15.x` became available

Debug logs for `@types/node` in `backend/package.json` (The PR for upgrading to `14.17.0` got opened as expected):
```json
{
  "depType": "devDependencies",
  "depName": "@types/node",
  "currentValue": "14.14.45",
  "datasource": "npm",
  "prettyDepType": "devDependency",
  "lockedVersion": "14.14.45",
  "depIndex": 12,
  "updates": [
    {
      "bucket": "minor",
      "newVersion": "14.17.0",
      "newValue": "14.17.0",
      "releaseTimestamp": "2021-05-18T17:05:22.569Z",
      "newMajor": 14,
      "newMinor": 17,
      "updateType": "minor",
      "branchName": "renovate/node-14.x"
    },
    {
      "bucket": "major",
      "newVersion": "15.3.0",
      "newValue": "15.3.0",
      "releaseTimestamp": "2021-05-14T19:03:38.669Z",
      "newMajor": 15,
      "newMinor": 3,
      "updateType": "major",
      "branchName": "renovate/node-15.x"
    }
  ],
  "warnings": [],
  "sourceUrl": "https://github.com/DefinitelyTyped/DefinitelyTyped",
  "sourceDirectory": "types/node",
  "currentVersion": "14.14.45",
  "isSingleVersion": true,
  "fixedVersion": "14.14.45"
}

```
Debug logs for `@types/node` in `frontend/package.json`

```json
{
  "depType": "devDependencies",
  "depName": "@types/node",
  "currentValue": "12.20.13",
  "datasource": "npm",
  "prettyDepType": "devDependency",
  "lockedVersion": "12.20.13",
  "depIndex": 8,
  "updates": [
    {
      "bucket": "major",
      "newVersion": "15.3.0",
      "newValue": "15.3.0",
      "releaseTimestamp": "2021-05-14T19:03:38.669Z",
      "newMajor": 15,
      "newMinor": 3,
      "updateType": "major",
      "branchName": "renovate/node-15.x"
    }
  ],
  "warnings": [],
  "sourceUrl": "https://github.com/DefinitelyTyped/DefinitelyTyped",
  "sourceDirectory": "types/node",
  "currentVersion": "12.20.13",
  "isSingleVersion": true,
  "fixedVersion": "12.20.13"
}
```
After this, there is no mention at all anymore of `node-15.x`.

Looking at the `branchName` in these logs (which is the same for the node docker image updates) makes me think that these is some sort of conflict somewhere that might prevent these PRs from being opened.

Is there some sort of config option I'm missing to make this work?

**Comment by rarkins** [comment]
You could:
- Check for an existing merged or closed PR and see if there is a comment on it saying it's blocking a new PR, or
- Check the debug logs to check why the branch isn't being created, or
- Using the Dependency Dashboard to see if the branch shows up as Blocked/Ignored and tick its checkbox

**Comment by viceice** [comment, marked as answer]
If I remember correctly major updates for types are disabled for npm types packages in `config:base` preset.

**Reply by lauraseidler** [reply]
Indeed - actually only for `@types/node`: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/helpers.ts#L4-L12

I'm not sure I understand the reason for this, since node major version updates are _not_ disabled by default AFAICS. 

My proposal would be at least to add this to the documentation for [JavaScript](https://docs.renovatebot.com/javascript/), similar as for [Docker](https://docs.renovatebot.com/docker/#enable-docker-major-updates). 

Maybe a preset `helper:enableTypesNodeMajor` would also be helpful then (`npm:enableTypesNodeMajor` would make more sense IMO, but that would be inconsistent)? 

I'm happy to raise a PR if either or both are an option.

**Reply by rarkins** [reply]
If we were to use `node` versioning for `@types/node` then maybe this rule could be removed?

**Reply by lauraseidler** [reply]
Would make sense to me - maybe this should also apply for docker node tags (afaik it doesn't)?

**Reply by rarkins** [reply]
There is this preset: https://github.com/renovatebot/renovate/blob/52e694bec9f2100fdd921903af0ef76ff5f44183/lib/config/presets/internal/helpers.ts#L4-L12

But I don't see it included by default anywhere, so they shouldn't be disabled by default as you mention. There's no need for a preset to enable them if by default everything is enabled anyway.

It's very surprising if `"renovate/node-15.x"` is not mentioned in the logs anywhere after an update references it.

**Reply by lauraseidler** [reply]
It's included in `config:base`: 

https://github.com/renovatebot/renovate/blob/5ef3b3ee05e0e44d1f379a9c1f66d05d7e2b63e2/lib/config/presets/internal/config.ts#L20

**Reply by viceice** [reply]
you can add it to `ignorePresets` as workaround 😄 

https://github.com/visualon/renovate-config/blob/004e24cb4c052c6d18d0c4c6c7182cc9019a9263/shared.json#L19

**Reply by lauraseidler** [reply]
oh interesting - I just duplicated the package rule and set it to enabled, but this is even nicer. 

I still feel like it's rather not intuitive that this is set by default tho, so I'd at least include a note in the documentation (like for docker)

**Reply by rarkins** [reply]
Addressed in #10085

**Reply by lauraseidler** [reply]
woah nice! thanks for the quick responses, really appreciate it!
